libraries/net/node.cpp

```diff
@@ -3135,7 +3155,11 @@ namespace graphene { namespace net { namespace detail {
       {
         if (firewall_check_state->expected_node_id != peer->node_id && // it's not the node who is asking us to test
             !peer->firewall_check_state && // the peer isn't already performing a check for another node
-            firewall_check_state->nodes_already_tested.find(peer->node_id) == firewall_check_state->nodes_already_tested.end() &&
+            _address_builder != nullptr &&
+            peer->get_remote_endpoint().valid() && 
```

Comment:
Is there a scenario that it's not valid?

Reply:
There were actually two instances of this validity check, both based on entries in the _active_connections list. There is no reason to check this, as there is no way an invalid endpoint could be placed in the collection.

Reply:
Where is the code that initializes the `remote_endpoint` variable?
IIRC there is code to update a remote endpoint's IP address from an internal adreee to an external address. Not sure if it's related.